server/mergin/app.py

```diff
@@ -105,9 +107,24 @@ def update_obj(self, obj):
                 field.populate_obj(obj, name)
 
 
-def create_simple_app() -> Flask:
-    from .config import Configuration
+class GeventTimeoutMiddleware:
+    """Middleware to implement gevent.Timeout() for all requests"""
+
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        request = Request(environ)
+        try:
+            with gevent.Timeout(Configuration.GEVENT_REQUEST_TIMEOUT):
+                return self.app(environ, start_response)
+        except gevent.Timeout:
+            logging.error(f"Gevent worker: Request {request.path} timed out")
+            resp = Response("Bad Gateway", mimetype="text/plain", status=502)
```

Comment:
Would a [504](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504) not make more sense here?

Reply:
done